axelor-budget/src/main/java/com/axelor/apps/budget/web/PurchaseOrderLineController.java

```diff
@@ -254,4 +259,46 @@ public void setBudgetDomain(ActionRequest request, ActionResponse response) {
       TraceBackService.trace(response, e);
     }
   }
+
+  public void setAnalyticDistributionPanelHidden(ActionRequest request, ActionResponse response) {
+    try {
+      PurchaseOrder purchaseOrder =
+          ContextTool.getContextParent(request.getContext(), PurchaseOrder.class, 1);
+
+      if (purchaseOrder == null || purchaseOrder.getCompany() == null) {
+        return;
+      }
+
+      PurchaseOrderLine purchaseOrderLine = request.getContext().asType(PurchaseOrderLine.class);
+      AnalyticLineBudgetModel analyticLineBudgetModel =
+          new AnalyticLineBudgetModel(purchaseOrderLine, purchaseOrder);
+      Map<String, Map<String, Object>> attrsMap = new HashMap<>();
+
+      Beans.get(AnalyticAttrsSupplychainService.class)
+          .addAnalyticDistributionPanelHiddenAttrs(analyticLineBudgetModel, attrsMap);
+      response.setAttrs(attrsMap);
+    } catch (Exception e) {
+      TraceBackService.trace(response, e, ResponseMessageType.ERROR);
+    }
+  }
+
+  public void accountOnChange(ActionRequest request, ActionResponse response) {
```

Comment:
I don't think that for `PurchaseOrderLine` there should be controller methods called something like `accountOnChange` as there can be more than one on the `onChange`